app/src/main/java/org/wikipedia/talk/TalkTopicActivity.kt

```diff
@@ -356,18 +370,22 @@ class TalkTopicActivity : BaseActivity(), LinkPreviewDialog.Callback {
                 }
                 .observeOn(AndroidSchedulers.mainThread())
                 .subscribe({
-                    onSaveSuccess(it)
+                    onSaveSuccess(it, showUndoSnackbar)
                 }, { t ->
                     L.e(t)
                     onSaveError(t)
                 }))
     }
 
-    private fun onSaveSuccess(newRevision: Long) {
+    private fun onSaveSuccess(newRevision: Long, showUndoSnackbar: Boolean) {
         binding.talkProgressBar.visibility = View.GONE
         binding.replySaveButton.isEnabled = true
         editFunnel.logSaved(newRevision)
 
+        if (showUndoSnackbar) {
+            showUndoSnackbar()
+        }
+
         if (isNewTopic()) {
             setResult(RESULT_EDIT_SUCCESS)
             finish()
```

Comment:
What happens if this branch is reached? i.e. if the user creates a "new topic" instead of replying to an existing topic?

Reply:
Done!

Reply:
@cooltey I'm not sure this is resolved:  to clarify, we _should_ allow the user to undo a new topic.

Reply:
Got it. Done!